st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -579,13 +579,39 @@ def _connect(self, host, socket=None):
         if socket:
             conninfo['sock'] = socket
 
+        conninfo = self._get_conninfo_from_ssh_config(conninfo)
         client = paramiko.SSHClient()
         client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
         client.connect(**conninfo)
 
         return client
 
     @staticmethod
+    def _get_conninfo_from_ssh_config(conninfo):
```

Comment:
Yeah, what lakshmi has said - In fact, I would prefer this method to just retrieve info from the SSH config and return a dictionary and the merging to happen somewhere else.